components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -2411,8 +2420,21 @@ public static API addAPIWithGeneratedSwaggerDefinition(API apiToAdd, String oasV
             }
             apiToAdd.setSwaggerDefinition(apiDefinition);
         } else {
-            AsyncApiParser asyncApiParser = new AsyncApiParser();
+            /**
+             * Note: By default, AsyncAPI version 3.0 will be used when creating new streaming APIs
+             * by defining the AsyncApiParser as AsyncApiV3Parser.
+             * If you need to use AsyncAPI version 2.x instead, use the following line
+             * AsyncApiParser asyncApiParser = AsyncApiParserFactory.getAsyncApiParser(
+             * APISpecParserConstants.AsyncApi.ASYNC_API_V2, getConfiguredDefaultParser());
+             */
+            AbstractAsyncApiParser asyncApiParser = AsyncApiParserFactory.getAsyncApiParser(
+                    APISpecParserConstants.AsyncApi.ASYNC_API_V3, getConfiguredDefaultParser());
             String asyncApiDefinition = asyncApiParser.generateAsyncAPIDefinition(apiToAdd);
+            if (log.isDebugEnabled()) {
+                String preview = asyncApiDefinition.length() > 100 ? asyncApiDefinition.substring(0, 100)
+                        + "...(truncated)" : asyncApiDefinition;
+                log.debug("Generated Swagger/AsyncApi Definition content: " + preview);
```

Comment:
we dont need this log. this will add noise when debugging any other issue in class